service/src/main/java/bio/terra/workspace/service/workspace/flight/create/azure/CreateDbAzureCloudContextFinishStep.java

```diff
@@ -15,20 +20,34 @@
 public class CreateDbAzureCloudContextFinishStep implements Step {
   private final UUID workspaceUuid;
   private final AzureCloudContextService azureCloudContextService;
+  private final FeatureConfiguration featureConfiguration;
 
   public CreateDbAzureCloudContextFinishStep(
-      UUID workspaceUuid, AzureCloudContextService azureCloudContextService) {
+      UUID workspaceUuid,
+      AzureCloudContextService azureCloudContextService,
+      FeatureConfiguration featureConfiguration) {
     this.workspaceUuid = workspaceUuid;
     this.azureCloudContextService = azureCloudContextService;
+    this.featureConfiguration = featureConfiguration;
   }
 
   @Override
   public StepResult doStep(FlightContext flightContext) throws InterruptedException {
-    AzureCloudContext azureCloudContext =
-        flightContext
-            .getInputParameters()
-            .get(JobMapKeys.REQUEST.getKeyName(), AzureCloudContext.class);
-
+    AzureCloudContext azureCloudContext;
+    if (featureConfiguration.isBpmEnabled()) {
+      azureCloudContext =
```

Comment:
Wiring in the values from BPM as the cloud context values for azure managed resource groups. I've marked the azure cloud context tenant Id, sub ID and resource group ID as deprecated on the cloud context creation call due to this change.

Reply:
You can see that if we serialized the spend profile, we could evolve it without changing this flight.